Fails to build on NixOS

```
trace: warning: crane will use a placeholder value since `name` cannot be found in /nix/store/61n0iniayircs79wwcafkvmiwvh7xnj9-Cargo.toml
to silence this warning consider one of the following:
- setting `pname = "...";` in the derivation arguments explicitly
- setting `package.name = "..."` or `workspace.package.name = "..."` in the root Cargo.toml
- explicitly looking up the values from a different Cargo.toml via 
  `craneLib.crateNameFromCargoToml { cargoToml = ./path/to/Cargo.toml; }`

To find the source of this warning, rerun nix with:
`NIX_ABORT_ON_WARN=1 nix --option pure-eval false --show-trace ...` 

error: builder for '/nix/store/60cnx7qa0m7zcq4y55n10wwg6f6xmwxa-cargo-package-deps-0.3.18.drv' failed with exit code 101;
       last 18 log lines:
       > cargoArtifacts not set, will not reuse any cargo artifacts
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/z51qfc9k0sdc106dkqy230q3pidhl59j-source
       > source root is source
       > Running phase: patchPhase
       > Executing configureCargoCommonVars
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > will append /build/source/.cargo-home/config.toml with contents of /nix/store/b5n97sd6z5yfy4jqb4xkirkca5nck12v-vendor-cargo-deps/config.toml
       > default configurePhase, nothing to do
       > Running phase: buildPhase
       > ++ command cargo --version
       > cargo 1.77.1
       > ++ command cargo check --release --locked --all-targets
       > error: failed to parse lock file at: /build/source/Cargo.lock
       >
       > Caused by:
       >   lock file version 4 requires `-Znext-lockfile-bump`
       For full logs, run 'nix log /nix/store/60cnx7qa0m7zcq4y55n10wwg6f6xmwxa-cargo-package-deps-0.3.18.drv'.
error: 1 dependencies of derivation '/nix/store/fp2pkd0xiiy3nircb5mahbpf36wq10z4-cargo-package-0.3.18.drv' failed to build
error: 1 dependencies of derivation '/nix/store/b2cpnm9vnn2jk6xcbmrn02ag4kyph55z-cargo-package-0.3.18_fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1a9h5kx65pcmk11naghvak3ral8013i1-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vabkwcabs8jnapgqlkcwdczfgh780v58-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7sqjz8mj2ani3d04nj759zjcskam61bb-nixos-system-marty-pc-25.05.20250218.73cf49b.drv' failed to build
```

System is on nixpkgs and nixos unstable, the flake.nix of this github repo specifies 24.10, might this be the reason?
I'm really rather new to NixOS and only very slowly starting to understand it, so no clue where to start.

Installing via cargo results in 
```
error[E0061]: this function takes 2 arguments but 1 argument was supplied
   --> /home/marty/.cargo/registry/src/index.crates.io-6f17d22bba15001f/distrans-0.3.1/src/app.rs:152:47
    |
152 | ..._) = new_routing_context(&self.cli.state_dir()?).await?;
    |         ^^^^^^^^^^^^^^^^^^^------------------------ argument #2 of type `Option<std::string::String>` is missing
    |
note: function defined here
   --> /home/marty/.cargo/registry/src/index.crates.io-6f17d22bba15001f/distrans_peer-0.3.18/src/lib.rs:24:14
    |
24  | pub async fn new_routing_context(
    |              ^^^^^^^^^^^^^^^^^^^
help: provide the argument
    |
152 |         let (routing_context, update_tx, _) = new_routing_context(&self.cli.state_dir()?, /* Option<std::string::String> */).await?;
    |                                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0061`.
error: could not compile `distrans` (lib) due to 1 previous error
error: failed to compile `distrans v0.3.1`, intermediate artifacts can be found at `/tmp/cargo-installXbgzMo`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

--- a/.github/workflows/flake.yml
+++ b/.github/workflows/flake.yml
@@ -1,0 +1,18 @@
+name: "Nix Flake"
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: cachix/install-nix-action@v27
+      with:
+        github_access_token: ${{ secrets.GITHUB_TOKEN }}
+    - run: nix build
+    - run: nix flake check

--- a/flake.nix
+++ b/flake.nix
@@ -1,31 +1,27 @@
 {
-  description = "ddcp";
+  description = "distrans";
 
   inputs = {
-    # NixOS 23.11 has recent enough versions of capnproto and protobuf to
+    # NixOS 24.11 has recent enough versions of capnproto and protobuf to
     # develop on Veilid.
-    nixpkgs.url = "github:nixos/nixpkgs/nixos-24.05";
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-24.11";
 
     rust-overlay = {
       url = "github:oxalica/rust-overlay";
       inputs.nixpkgs.follows = "nixpkgs";
-      inputs.flake-utils.follows = "flake-utils";
     };
 
     crane.url = "github:ipetkov/crane";
-    crane.inputs.nixpkgs.follows = "nixpkgs";
   };
 
   outputs = { self, nixpkgs, flake-utils, rust-overlay, crane }:
     (flake-utils.lib.eachDefaultSystem (system:
       let
+        overlays = [ (import rust-overlay) ];
         pkgs = import nixpkgs {
-          inherit system;
-          overlays = [
-            rust-overlay.overlays.default
-          ];
+          inherit system overlays;
         };
-
+        craneLib = crane.mkLib pkgs;
         arch = flake-utils.lib.system.system;
 
       in {
@@ -34,7 +30,7 @@
           buildInputs = (with pkgs; [
             cargo
             cargo-watch
-            (rust-bin.stable.latest.default.override { extensions = [ "rust-src" ]; })
+            (rust-bin.stable."1.81.0".default.override { extensions = [ "rust-src" ]; })
             rustfmt
             rust-analyzer
             clang
@@ -52,12 +48,13 @@
           LIBCLANG_PATH="${pkgs.llvmPackages.libclang.lib}/lib";
         };
 
-        packages.default = crane.lib.${system}.buildPackage {
+        packages.default = craneLib.buildPackage {
+          pname = "distrans";
           src = ./.;
 
           buildInputs = with pkgs; [
             cargo
-            rust-bin.stable.latest.default
+            rust-bin.stable."1.81.0".default
             capnproto
             protobuf
             pkg-config

